marlowe-playground-client/src/MainFrame/State.purs

```diff
@@ -322,26 +351,32 @@ handleAction s (HandleBlocklyMessage (CurrentCode code)) = do
   --   void $ query _blocklySlot unit (Blockly.SetError "You can't send new code to a running simulation. Please go to the Simulation tab and click \"reset\" first" unit)
   -- else do
   selectView Simulation
-  void $ toSimulation $ Simulation.handleAction s (ST.SetEditorText code)
+  void $ toSimulation $ Simulation.handleAction settings (ST.SetEditorText code)
+
+handleAction settings (HandleBlocklyMessage _) = pure unit
 
 handleAction _ (HandleActusBlocklyMessage ActusBlockly.Initialized) = pure unit
 
-handleAction s (HandleActusBlocklyMessage (ActusBlockly.CurrentTerms flavour terms)) = do
+handleAction settings (HandleActusBlocklyMessage (ActusBlockly.CurrentTerms flavour terms)) = do
   let
     parsedTermsEither = AMB.parseActusJsonCode terms
   case parsedTermsEither of
     Left e -> void $ query _actusBlocklySlot unit (ActusBlockly.SetError ("Couldn't parse contract-terms - " <> (show e)) unit)
     Right parsedTerms -> do
       result <- case flavour of
-        ActusBlockly.FS -> runAjax $ flip runReaderT s $ (Server.postApiActusGenerate parsedTerms)
-        ActusBlockly.F -> runAjax $ flip runReaderT s $ (Server.postApiActusGeneratestatic parsedTerms)
+        ActusBlockly.FS -> runAjax $ flip runReaderT settings $ (Server.postApiActusGenerate parsedTerms)
+        ActusBlockly.F -> runAjax $ flip runReaderT settings $ (Server.postApiActusGeneratestatic parsedTerms)
       case result of
         Success contractAST -> do
           selectView Simulation
-          void $ toSimulation $ Simulation.handleAction s (ST.SetEditorText contractAST)
+          void $ toSimulation $ Simulation.handleAction settings (ST.SetEditorText contractAST)
         Failure e -> void $ query _actusBlocklySlot unit (ActusBlockly.SetError ("Server error! " <> (showErrorDescription (runAjaxError e).description)) unit)
         _ -> void $ query _actusBlocklySlot unit (ActusBlockly.SetError "Unknown server error!" unit)
 
+-- QUESTION: How does this relates to the handleGistAction?
+-- the save the project we use the PublishGist action and I thought that when we loaded
+-- we used the LoadGist action, but it's actually this one.
```

Comment:
The issue is that for historical reasons, the gist actions rely on gist id stored in the state, so we need to set the appropriate state before handling the gist action. This should probably be changed to have gist action type taking gist id as a parameter.